eta/core/datasets.py

```diff
@@ -738,31 +743,34 @@ def add_file(self, data_path, labels_path, file_method=COPY,
                              % os.path.basename(data_path))
 
         data_subdir = os.path.join(self.data_dir, self._DATA_SUBDIR)
-        new_data_file = os.path.basename(data_path)
+        if new_data_filename is None:
+            new_data_filename = os.path.basename(data_path)
 
         labels_subdir = os.path.join(self.data_dir, self._LABELS_SUBDIR)
-        new_labels_file = os.path.basename(labels_path)
+        if new_labels_filename is None:
+            new_labels_filename = os.path.basename(data_path)
```

Comment:
Won't this mean that the labels file will have the same extension as the data file (e.g. `.mp4`, `.jpg`), instead of `.json`?

Reply:
I tested it; it does indeed give the labels file the same extension as the data file.  I think this line should read:
```python
new_labels_filename = os.path.basename(labels_path)
```